GCA Intern: Improve DMARC and SPF advice

SeanM-temp opened the pull request:
Adds additional checks for DMARC and SPF records. 
DMARC
-Check for v=DMARC1
-Check for p= tag
-Verify akim and aspf values
SPF
-Warning for ptr mechanism
-Check for exceeding dns lookup limit

SeanM-temp committed:
Update advisor.go

Improved DMARC and SPF advice

SeanM-temp committed:
DMARC fix and comment

wolveix committed:
Overhauled existing tests
Mapped 100% of the DMARC advisor's function with tests
Cleaned up advice formatting

SeanM-temp committed:
Added recursive lookup to SPF advice

Added check and advice for DNS lookup limit and cyclic lookup.

SeanM-temp committed:
Merge branch 'main' into SPF-recursive-lookup

SeanM-temp committed:
Merge pull request #1 from SeanM-temp/SPF-recursive-lookup

Added recursive lookup to SPF advice

wolveix commented:
Thanks for the changes, @SeanM-temp!

The existing separation of the `advisor` and `scanner` packages does make validating the number of SPF host lookups difficult. The `advisor` package shouldn't import `scanner`; however, the custom DNS implementation being outside of the `scanner` package is difficult. I'll find some time to properly look over your implementation, and see whether we can improve it :)

SeanM-temp committed:
-Moved DNS scanning to DNS package
-Merged Advisor and Scanner packages into Scanner package
-Added MTA-STS scanning and advice
-Added DNSSEC scanning
-Improved STS advice

SeanM-temp committed:
Removed redundant Advisor package

SeanM-temp committed:
Removed commented out code

SeanM-temp committed:
Updated README.md to include --dnssec

wolveix commented:
Hey @SeanM-temp, apologies that this hasn't been properly reviewed and/or merged yet! I intend to look over it soon :)

wolveix committed:
Always scan DNSSEC

wolveix committed:
Backport several changes

wolveix committed:
Merge remote-tracking branch 'origin/main' into fork/SeanM-temp/main

# Conflicts:
#	cmd/dss/main.go
#	cmd/dss/scan.go
#	cmd/dss/serve.go
#	pkg/http/server.go
#	pkg/mail/server.go
#	pkg/mail/template.go
#	pkg/model/scan.go
#	pkg/scanner/requests.go
#	pkg/scanner/scanner.go

wolveix committed:
Cleanup

wolveix committed:
Further cleanup
